tensorflow/core/kernels/mkl_fused_ops_test.cc

```diff
@@ -618,87 +617,137 @@ template <typename T>
-                           .Input(FakeInput(DT_UINT8))
-                           .Input(FakeInput(num_args, DT_UINT8))
-                           .Attr("T", dtype)
-                           .Attr("transpose_a", false)
-                           .Attr("transpose_b", false)
-                           .Attr("num_args", num_args)
-                           .Attr("fused_ops", fused_ops)
-                           .Attr("epsilon", 0.0001)
-                           .Attr("_kernel", "MklLayoutDependentOp")
-                           .Finalize(node_def()));
-
-          TF_EXPECT_OK(InitOp());
-
-          AddInputFromArray<T>(input.shape(), input.flat<T>());
-          AddInputFromArray<T>(weight.shape(), weight.flat<T>());
-          AddInputFromArray<T>(bias.shape(), bias.flat<T>());
-          // Add MKL meta input for input, filter and bias.
-          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-          AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-
-          TF_ASSERT_OK(RunOpKernel());
-
-          const Tensor& output_tensor = *GetOutput(0);
-          const Tensor& output_meta_tensor = *GetOutput(1);
-          CommonTestUtilities<T> test_util;
-          test_util.PerformConversion(dtype, output_tensor, output_meta_tensor,
-                                      output);
-        };
+                         const int kOutputChannel,
+                         const std::vector<string>& fused_ops) {
+    const FusedGraphRunner run_default = [this](
+        const Tensor& input, const Tensor& weight, const Tensor& bias,
+        const std::vector<string>& fused_ops, Tensor* output) {
+      auto root = tensorflow::Scope::NewRootScope();
+      auto input_op =
+          ops::Const(root.WithOpName("input"), Input::Initializer(input));
+      Output next_op = ops::MatMul(
+          root.WithOpName("matmul"), input_op,
+          ops::Const(root.WithOpName("weight"), Input::Initializer(weight)));
+
+      string last_op = "";
+      if (std::find(fused_ops.begin(), fused_ops.end(), "BiasAdd") !=
+          fused_ops.end()) {
+        last_op = "with_bias";
+        next_op = ops::BiasAdd(
+            root.WithOpName(last_op), next_op,
+            ops::Const(root.WithOpName("bias"), Input::Initializer(bias)));
+      }
+
+      if (std::find(fused_ops.begin(), fused_ops.end(), "Relu") !=
+          fused_ops.end()) {
+        last_op = "with_relu";
+        next_op = ops::Relu(root.WithOpName(last_op), next_op);
+      }
+
+      if (std::find(fused_ops.begin(), fused_ops.end(), "Relu6") !=
+          fused_ops.end()) {
+        last_op = "with_relu6";
+        next_op = ops::Relu6(root.WithOpName(last_op), next_op);
+      }
+
+      if (std::find(fused_ops.begin(), fused_ops.end(), "Elu") !=
+          fused_ops.end()) {
+        last_op = "with_elu";
+        next_op = ops::Relu(root.WithOpName(last_op), next_op);
```

Comment:
Thank for your reviewing, Relu will have the same result with Elu when the activation is greater than 0. I checked the code and found that the test always generate positive float so that this error was hidden after testing. I'll correct it.

Reply:
Fixed.